69.md

```diff
@@ -0,0 +1,293 @@
+  },
+  "nip69": {
+    "bob": "noffer1..."
+  }
+}
+```
+
+### NIP-57 Zaps
+
+Instead of using the LNURL-pay callback, the "zap" flow can be initiated as content payload to an Offer:
+
+1. The client creates a kind 9734 zap request event as specified in NIP-57.
+2. Instead of sending this to an LNURL-pay endpoint, the client sends it as the `zap` part of the encrypted content in a NIP-69 payment request.
+3. The receiver processes the NIP-69 payment request, decrypts the content, extracts the kind 9734 event, and uses it to generate the invoice.
+4. The receiver responds with the invoice as specified in NIP-69.
+5. Once paid, the receiver generates and publishes the kind 9735 zap receipt as specified in NIP-57.
+
+Example NIP-57 zap request:
+
+```json
+{
+  "id": "<event_id>",
+  "pubkey": "<sender_pubkey>",
+  "created_at": 1234567890,
+  "kind": 21001,
+  "tags": [
+    ["p", "<receiver_pubkey>"]
+  ],
+  "content": "<NIP-44 encrypted {\"offer\":\"<zap1234>\",\"zap\":<kind 9734 zap request event>}>",
+  "sig": "<signature>"
+}
+```
+
+There is no inherent requirement on the service to acknowledge zap requests, since spontaneous payments are default behavior. However, both Clients and Service implementations SHOULD consider offer identifiers, where `startsWith("zap")`, indication that the service will honor zap requests for that offer. A service should not error if no zap request was included by the sender on a zap indicated offer, the service should treat it as any other spontaneous payment.
+
+## General Process Flow
+
+1. **Payer Scans or Clicks the Static Payment Code**
+2. **Payer's Wallet Decodes the Payment Code**
+3. **Payer's Wallet Sends a Nostr Event to the Specified Relay**
+   - Addressed to the receiver's public key, containing the offer identifying string.
+   - Event Type: Ephemeral Kind 21001 | NIP-44 Encrypted
+   - **Conditional**: Include the amount in sats the sender wishes to pay if the payment type is spontaneous.
+   - Optional: Include additional payer data.
+4. **Receiver Responds with Lightning Invoice**
+   - Generates a Lightning invoice and responds with a Nostr event containing the invoice details.
+   - Event Type: Ephemeral Kind 21001 | NIP-44 Encrypted
+   - **Content**: `{"bolt11":"<BOLT11_invoice_string>"}`
+   - Optional: Include additional purchase data.
+5. **Payer Pays the Invoice**
+   - Completes the payment by settling the Lightning invoice.
+6. **Optional: Receiver Emits Payment Receipt**
+   - This step is out of scope for this NIP, but potential receipt considerations include:
+     - Nostr Event as Receipt: Emitting a public Nostr event as a "zap" receipt for verifiable record-keeping.
+     - Lightning Pre-Image: The Lightning pre-image is proof of payment.
+     - External / WebHook: The receiver may finalize the flow out-of-band.
+
+### Nostr Events
+
+This NIP specifies the use of event kind `21001` with the following structure:
+
+- `content`: NIP-44 encrypted payment details.
```

Comment:
Is it worth gift-wrapping these events (NIP-59/17) to increase privacy and reduce metadata leaks? How are the privacy comparisons here to bolt-12 with blinded paths? Is the thinking that the sender could always just use a new anonymous keypair for a payment if they want to?

Reply:
Exactly yea... for those not also on Telegram:

I think its much better, bolt12 has to onion things because its correlating the comms and the lightning node, which with the combined metadata can reveal a lot

With nostr, the comms are separate from the Lightning Node, a nostr key could service many nodes or one lightning node could have many nostr keys.

The nostr keys for applications/wallet services aren't inherently linked to social either, they're meant to be quasi-ephemeral. The giftwrapping/seal nips don't quite make sense to me as far as solving the handshake problem, even simplex is just doing that out of band with ephemeral keys

Reply:
Agree, I don't think that level of privacy is needed. But if you are concerned with relays tracking users by IP then you could use the gift wrapping spec to route the message through a second trusted relay or DVM as a native onion routing technique.